packages/host/app/resources/metamask.ts

```diff
@@ -0,0 +1,111 @@
+import { tracked } from '@glimmer/tracking';
+import { Resource } from 'ember-resources/core';
+import { enqueueTask, restartableTask } from 'ember-concurrency';
+
+// window is a global object that is available in this resource
+declare global {
+  interface Window {
+    ethereum: any;
+  }
+}
+
+const METAMASK_ERROR_CODES = {
+  user_rejected: 4001,
+  unknown_chain: 4902,
+};
+
+class MetaMaskResource extends Resource {
+  @tracked connected = false;
+  @tracked chainId = -1; // the chain id of the metamask connection (not the card)
+
+  setup() {
+    this.doInitialize.perform();
+    if (this.isMetamaskInstalled()) {
+      window.ethereum.on('chainChanged', this.handleChainChanged.bind(this));
+    }
+    return this;
+  }
+
+  private doInitialize = enqueueTask(async () => {
+    if (this.isMetamaskInstalled()) {
+      let chainId = this.getChainId();
+      let connected = await this.isMetamaskConnected();
+      this.chainId = chainId;
+      this.connected = connected;
+    }
+  });
+
+  teardown() {
```

Comment:
I don't think anything is calling this.

Reply:
Right yea. I just added a destroy hook here https://github.com/cardstack/boxel/pull/372/commits/6acd1fdbdf62c30486fa63cded2e665f3b8c7cdd